### ISSUE TITLE
Schema doesn't serialize PnP:Provider elemenets

### ISSUE THREAD

**Issue by donaldhessing** [OfficeDev/PnP-Provisioning-Schema#51]
Played with the extensibility providers but can't get this working via declarative XML. 
1. Added the PnP:Providers elements in the template.xml. After loading the template.Providers is null
2. Added the template.Providers via object model. This works just fine - extensibility provider is loaded and the ProcessRequest method is invoked. When I save the template object to XML i can see the correct PnP:Providers elements. However, after loading the saved xml template, the template.Providers object is again null. So,this works just fine if the provider is added via the object model, but doesn't work if the provider is added via in the template.xml file.

**Comment by PaoloPia** [comment]
Hi Donald,
Thanks for your feedback. Actually, the formatter should support custom providers. Can you share with me the XML template that you are using? Just to double-check ...

Thanks.

**Comment by donaldhessing** [comment]
Hi Paolo,
 
Sure - It contains some customer information but trust that is not an issue.
 
Attached files:
- CanonProvisioningTemplate.xml -> this xml file is first extracted from an existing site. I then added the PnP:Providers element based on the blog from Erwin
- PnPProviderDemo.xml -> this is a saved xml file from the template object (provider.SaveAs()) in which I added the provider objects via code. This works as expected, but it only works by adding the objects in code. If I load the saved xml template the template object doesn't have a collection for providers.
 
Many thanks,
Donald
 
 

 
Date: Tue, 1 Dec 2015 03:36:18 -0800
